(Doc+) link videos for allocation and ilm (#116880)
elastic/elasticsearch: docs/reference/cluster/allocation-explain.asciidoc

<5> The decider which led to the `no` decision for the node.
<6> An explanation as to why the decider returned a `no` decision, with a helpful hint pointing to the setting that led to the decision. In this example, a newly created index has <<indices-get-settings,an index setting>> that requires that it only be allocated to a node named `nonexistent_node`, which does not exist, so the index is unable to allocate.

> See this https://www.youtube.com/watch?v=5z3n2VgusLE[fixing "node does not match index setting" video] for an example troubleshooting walkthrough.

Reviewer: "example" and "walkthrough" are kind of the same thing. we also don't need to provide the title of the video in the link.

```suggestion
See https://www.youtube.com/watch?v=5z3n2VgusLE[this video] for a walkthrough of troubleshooting a node and index setting mismatch.
```

----
// NOTCONSOLE

> If a shard is unassigned with an allocation status of `no_valid_shard_copy`, then you should <<fix-cluster-status-recover-nodes,make sure that all nodes are in the cluster>>. If all the nodes containing in-sync copies of a shard are lost, then you can <<fix-cluster-status-restore,recover the data for the shard>>. See this https://www.youtube.com/watch?v=6OAg9IyXFO4[fixing "no_valid_shard_copy" video] for an example troubleshooting walkthrough.

Reviewer: should be its own paragraph so it can be seen
```suggestion
If a shard is unassigned with an allocation status of `no_valid_shard_copy`, then you should <<fix-cluster-status-recover-nodes,make sure that all nodes are in the cluster>>. If all the nodes containing in-sync copies of a shard are lost, then you can <<fix-cluster-status-restore,recover the data for the shard>>. 

See https://www.youtube.com/watch?v=6OAg9IyXFO4[this video] for a walkthrough of troubleshooting `no_valid_shard_copy`.
```
